libraries/core_libs/consensus/src/vote_manager/vote_manager.cpp

```diff
@@ -537,26 +544,44 @@ std::optional<PbftRound> VoteManager::determineNewRound(PbftPeriod current_pbft_
 }
 
 PbftPeriod VoteManager::getRewardVotesPbftBlockPeriod() {
-  std::shared_lock lock(reward_votes_info_mutex_);
+  std::shared_lock lock(verified_votes_access_);
   return reward_votes_period_;
 }
 
-void VoteManager::resetRewardVotesInfo(PbftPeriod period, PbftRound round, const blk_hash_t& block_hash) {
-  {
-    std::scoped_lock lock(reward_votes_info_mutex_);
-
-    reward_votes_block_hash_ = block_hash;
-    reward_votes_period_ = period;
-    reward_votes_round_ = round;
+void VoteManager::resetRewardVotes(PbftPeriod period, PbftRound round, PbftStep step, const blk_hash_t& block_hash,
+                                   DbStorage::Batch& batch) {
+  // Save 2t+1 cert votes to database, remove old reward votes
+  std::scoped_lock lock(verified_votes_access_);
+  reward_votes_block_hash_ = block_hash;
+  reward_votes_period_ = period;
+  reward_votes_round_ = round;
+
+  auto found_period_it = verified_votes_.find(period);
+  assert(found_period_it != verified_votes_.end());
+  auto found_round_it = found_period_it->second.find(round);
```

Comment:
segfault/or whatever would happen right here as found_period_it might be == end()....
Those asserts are ignored in release mode so you can use them but you have to use it like this:

```
auto found_period_it = verified_votes_.find(period);
if (found_period_it == verified_votes_.end()) {
  asset
  return
}
```

Reply:
Fixed